src/components/shared/FilterNavigatorBar.js

```diff
@@ -7,13 +7,17 @@
 import React, { PureComponent } from 'react';
 import classNames from 'classnames';
 import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { getFormattedTimeLength } from '../../profile-logic/range-filters';
 import './FilterNavigatorBar.css';
 
+import type { ProfileSelection } from '../../types/actions';
+
 type Props = {|
   +className: string,
   +items: string[],
   +onPop: number => *,
   +selectedItem: number,
+  +profileSelection: ProfileSelection,
```

Comment:
As said above, I believe this should be an optional string `uncommitedItem`, so you would write this like this:
```javascript
type Props = {|
  ...
  +uncommitedItem ?: string,
|};